const-generic-const-fn-bounds.md

```diff
@@ -131,6 +131,46 @@ It might also be desirable to make the automatic `Fn*` impls on function types a
 This change should probably go in hand with allowing `const fn` pointers on const functions
 that support being called (in contrast to regular function pointers).
 
+## Deriving `const impl`s
+
+```rust
+#[derive(Clone)]
+pub struct Foo(Bar);
+
+struct Bar;
+
+const impl Clone for Bar {
+    fn clone(&self) -> Self { Bar }
+}
+```
+
+could theoretically have a scheme inferring `Foo`'s `Clone` impl to be `const`. If some time
+later the `const impl Clone for Bar` (a private type) is changed to just `impl`, `Foo`'s `Clone`
+impl would suddenly stop being `const`, without any visible change to the API. This should not
+be allowed for the same reason as why we're not inferring `const` on functions: changes to private
+things should not affect the constness of public things, because that is not compatible with semver.
```

Comment:
Hmm, is the need for something like `#[derive(const Clone)]` mentioned anywhere explicitly?
I mean, this section looks like the motivation, but not the conclusion.

Reply:
Right, I just don't see `derive(const Foo)` mentioned at all.